libvast/src/data.cpp

```diff
@@ -483,6 +483,86 @@ bool convert(const caf::config_value& x, data& y) {
   return caf::visit(f, x);
 }
 
+record& put(record& r, std::string_view key, const duration& value) {
+  r[key] = value;
+  return r;
+}
+
+record& put(record& r, std::string_view key, const time& value) {
+  r[key] = value;
+  return r;
+}
+
+record& put(record& r, std::string_view key, std::string_view value) {
+  r[key] = std::string{value};
+  return r;
+}
+
+record& put(record& r, std::string_view key, const pattern& value) {
+  r[key] = value;
+  return r;
+}
+
+record& put(record& r, std::string_view key, const address& value) {
+  r[key] = value;
+  return r;
+}
+
+record& put(record& r, std::string_view key, const subnet& value) {
+  r[key] = value;
+  return r;
+}
+
+record& put(record& r, std::string_view key, const enumeration& value) {
+  r[key] = value;
+  return r;
+}
+
+record& put(record& r, std::string_view key, const list& value) {
+  r[key] = value;
+  return r;
+}
+
+record& put(record& r, std::string_view key, const map& value) {
+  r[key] = value;
+  return r;
+}
+
+record& put(record& r, std::string_view key, const record& value) {
+  r[key] = value;
+  return r;
+}
+
+record& put_record(record& r, std::string_view key) {
+  r[key] = record{};
+  return caf::get<record>(r[key]);
+}
+
+record& put_record(list& l) {
```

Comment:
How much effort do you think it would be to replace all usages of the put* functions so we don't need to provide these polyfills?